packages/sns/lib/sns/AbstractSnsSqsConsumer.ts

```diff
@@ -12,15 +12,15 @@ import { deleteSnsSqs, initSnsSqs } from '../utils/snsInitter'
 import { readSnsMessage } from '../utils/snsMessageReader'
 import type { SNSSubscriptionOptions } from '../utils/snsSubscriber'
 
-import type { SNSCreationConfig, SNSOptions, SNSQueueLocatorType } from './AbstractSnsService'
+import type { SNSCreationConfig, SNSOptions, SNSTopicLocatorType } from './AbstractSnsService'
 
 export type SNSSQSConsumerDependencies = SQSConsumerDependencies & {
   snsClient: SNSClient
 }
 export type SNSSQSCreationConfig = SQSCreationConfig & SNSCreationConfig
 
-export type SNSSQSQueueLocatorType = SQSQueueLocatorType &
-  SNSQueueLocatorType & {
+export type SNSSQSQueueLocatorType = Partial<SQSQueueLocatorType> &
+  SNSTopicLocatorType & {
```

Comment:
How configuration in billing service could look like?

Arguments passed to constructor should be wrapped with `isTest()` e.g.:
```
{
  "creationConfig": isTest() ? { /* legit creation config */ } : undefined, //In production creation of topics will be disabled.
  "locatorConfig": isTest() ? undefined : { /* legit locate config */ } //In production we will do read-only
}
```

Reply:
I'd say that creationConfig would always have an entry for the queue, and conditionally will include entry for the topic.
locatorConfig will conditionally either have an entry for the topic, or be empty

Reply:
Let me visualize it with the test. I think the rest of the comments will clarify too.